lib/dao/worker_dao.es6.js

```diff
@@ -0,0 +1,74 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+require('./indexed_dao.es6.js');
+require('./dao_factory.es6.js');
+
+foam.CLASS({
+  name: 'WorkerDAO',
```

Comment:
Do we use this in more than one place? Feels like a lot of bloat just to fetch + cache a URL?

Reply:
We use is multiple times in the same place. The hope is that the `api_service` will become something like:

```js
angular.service(/*.../ function() {
  return {
    releaseDAO: RestAndWorkerDAO.create({baseURLPath: `/releaseDAO`}),
    compatDAO: RestAndWorkerDAO.create({baseURLPath: `/compatDAO`}),
});
```

where `RestAndWorkerDAO` is a (not so great) name for the proxy that delegates directly to the REST API until the cached-in-worker DAO is ready. My main reason for "all the abstraction" here is to facilitate tests of each class with minimal baked in dependencies on code that isn't under test. This followed from me trying to think about how to make [this](https://github.com/mdittmer/confluence/blob/compat-catalog0/main/app2.es6.js) into testable parts of `api_service` + a UI that consumes the DAO.

I'm open to different approaches. Do you think all this wiring should just get baked in to `api_service` and we should write some kind of end-to-end test for `api_service` that somehow captures the network + caching behaviour?

Reply:
@lukebjerring PTAL at latest push. It eliminates the extra factory abstraction.